### PR TITLE
feat(protocol-designer): import modals and landing pg touch up

### PR DESCRIPTION
**Pull request by jerader**
closes partially AUTH-688 and AUTH-723

# Overview

Edits copy for landing page and adds the top level file import modal with different contents depending on the type.

## Test Plan and Hands on Testing

upload the following and observe the modals:

1. a non json file

3. an invalid json file (use example)
[Albumin ELISA P1 V2.0.json](https://github.com/user-attachments/files/16680857/Albumin.ELISA.P1.V2.0.json)

4. a file that needs a migration (use example)
[1.4 test (1) (1).json](https://github.com/user-attachments/files/16680878/1.4.test.1.1.json)

## Changelog

- add when modals pop up and fix copy of landing page
- add the modal with different content
- add tests

## Risk assessment

low

### CODE REVIEW

File: protocol-designer/src/NavigationBar.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ import {
 } from '@opentrons/components'
 import { actions as loadFileActions } from './load-file'
 import type { ThunkDispatch, RouteProps } from './types'
+import { getFileMetadata } from './file-data/selectors'
```
Comment (koji) [inline protocol-designer/src/NavigationBar.tsx]: this line can be higher

File: protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+export function FileUploadMessagesModal(): JSX.Element | null {
+  const message = useSelector(getFileUploadMessages)
+  const dispatch = useDispatch()
+  const { t } = useTranslation('shared')
+  const dismissModal = (): void => {
+    dispatch(dismissFileUploadMessage())
+  }
+  const modalContents = useFileUploadModalContents({
+    uploadResponse: message,
+  })
```
Comment (koji) [inline protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx]:
```suggestion
  const modalContents = useFileUploadModalContents({
    uploadResponse: message,
  })
  const dismissModal = (): void => {
    dispatch(dismissFileUploadMessage())
  }
```

File: protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  PrimaryButton,
+  SPACING,
+  SecondaryButton,
+} from '@opentrons/components'
+import { getFileUploadMessages } from '../../load-file/selectors'
+import { dismissFileUploadMessage, undoLoadFile } from '../../load-file/actions'
+import { useFileUploadModalContents } from './utils'
+
+export function FileUploadMessagesModal(): JSX.Element | null {
+  const message = useSelector(getFileUploadMessages)
```
Comment (koji) [inline protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx:17]:
```suggestion
  const { isError } = useSelector(getFileUploadMessages)
```

File: protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            <SecondaryButton
+              onClick={() => {
+                dispatch(undoLoadFile())
+              }}
+            >
+              {t('cancel')}
+            </SecondaryButton>
+            <PrimaryButton onClick={dismissModal}>{t('confirm')}</PrimaryButton>
+          </Flex>
+        ) : undefined
```
Comment (koji) [inline protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx]: if possible, `null` would be a better option for us.

File: protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+  return (
+    <Modal
+      type={message?.isError ? 'error' : 'info'}
+      title={title}
+      closeOnOutsideClick
+      onClose={dismissModal}
+      footer={
+        title !== t('invalid_json_file') &&
+        title !== t('incorrect_file_header') ? (
```
Comment (koji) [inline protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx]:
```suggestion
        !['invalid_json_file', 'incorrect_file_header'].includes(title)
```

File: protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx (hunk 4 above)
Comment (koji) [reply protocol-designer/src/organisms/FileUploadMessagesModal/index.tsx]: or use `&&` instead `? : `

File: protocol-designer/src/assets/localization/en/shared.json [hunk 6]
```diff
@@ -2,26 +2,65 @@
...
+      "body2": "Protocol Designer no longer supports aspirate or mix actions in a trash bin. If your protocol contains these actions, an error icon will appear next to them in the Protocol Timeline. To resolve the error, choose another location for aspirating or mixing.",
+      "body3": "Additionally, we have addressed a bug where blow out speeds were slower than expected. Your protocol will automatically update the flow rates unless they were specifically initialized.",
+      "body4": "As always, please contact us with any questions or feedback."
+    },
+    "generic": {
+      "body1": "Your protocol will be automatically updated to the latest version. Please note that the updated file will be incompatible with older versions of the Protocol Designer. We recommend making a separate copy of the file that you just imported before continuing.",
+      "body2": "Updating the file may make changes to liquid handling actions. Please review your file in the Protocol Designer",
+      "body3": "As always, please contact us with any questions or feedback."
+    },
+    "toV3Migration": {
```
Comment (ecormany) [inline protocol-designer/src/assets/localization/en/shared.json:50]: Is there any way to trigger this old migration? Are we just keeping it for posterity?

File: protocol-designer/src/assets/localization/en/shared.json [hunk 7]
```diff
@@ -2,26 +2,65 @@
...
   "done": "Done",
+  "edit_existing": "Edit existing protocol",
   "edit": "edit",
   "eight_channel": "8-Channel",
   "exit": "exit",
   "go_back": "Go back",
-  "import_existing": "Import existing protocol",
   "import": "Import",
+  "incorrect_file_header": "Invalid file type",
+  "incorrect_file_type_body": "Protocol Designer only accepts JSON files of protocols that were previously created in Protocol Designer. Upload a valid file to continue.",
```
Comment (ecormany) [inline protocol-designer/src/assets/localization/en/shared.json]: Would love to improve the wording here.

```suggestion
  "incorrect_file_type_body": "Protocol Designer only accepts JSON protocol files created with Protocol Designer. Upload a valid file to continue.",
```

File: protocol-designer/src/assets/localization/en/shared.json [hunk 8]
```diff
@@ -2,26 +2,65 @@
...
+  "invalid_json_file_body": "This file is either missing information it needs to import properly or contains sections that Protocol Designer cannot read. At this time we do not support JSON files created outside of Protocol Designer.",
```
Comment (ecormany) [inline protocol-designer/src/assets/localization/en/shared.json]: Another one we can tighten up.
```suggestion
  "invalid_json_file_body": "This JSON file is either missing required information or contains sections that Protocol Designer cannot read. At this time we do not support JSON files created outside of Protocol Designer.",
```

**Review by ecormany — COMMENTED** [review]
A couple copy suggestions but generally looking good on that front!

File: protocol-designer/src/assets/localization/en/shared.json (hunk 6 above)
Comment (jerader) [reply protocol-designer/src/assets/localization/en/shared.json:50]: the way the migrations work is the user will see only the modal contents for the first migration they go through. So if they have a very old protocol (like a v2) that goes through several migrations, they'll see only the v3 migration modal. IDK why we have it like that 😬 probably should be revisited so the user only sees the latest migration? But i want to keep current Pd behavior in here for now

Comment (ecormany) [reply protocol-designer/src/assets/localization/en/shared.json:50]: 💯 leave it in until we do that refactor.

**Review by koji — APPROVED** [review]
lgtm

**Pull request merged**